internal/cmd/commands/database/flags.go

```diff
@@ -0,0 +1,143 @@
+// - The flag for skipping the creation of the initial authenticated user role (`-skip-initial-authenticated-user-role-creation`) is provided.
+//
+// Returns:
+// - bool: True if the creation of the initial authenticated user role should be skipped, false otherwise.
+// - string: A reason string explaining why the creation is skipped, or an empty string if not skipped.
+func (f *initFlags) SkipInitialAuthenticatedUserRoleCreation() (bool, string) {
+	if f.flagSkipInitialAuthenticatedUserRoleCreation {
+		return true, reasonSkipInitialAuthenticatedUserRoleCreation
+	}
+	return false, ""
+}
+
+// SkipAuthMethodCreation checks if the creation of authentication methods should be skipped.
+// It returns a boolean indicating whether to skip and a reason string explaining why the creation is skipped.
+//
+// Authentication method creation is skipped if:
+// - The flag for skipping authentication method creation (`-skip-auth-method-creation`) is provided.
+//
+// Returns:
+// - bool: True if the creation of authentication methods should be skipped, false otherwise.
+// - string: A reason string explaining why the creation is skipped, or an empty string if not skipped.
+func (f *initFlags) SkipAuthMethodCreation() (bool, string) {
+	if f.flagSkipAuthMethodCreation {
+		return true, reasonSkipAuthMethodCreation
+	}
+	return false, ""
+}
+
+// SkipScopesCreation checks if the creation of default scopes should be skipped.
+// It returns a boolean indicating whether to skip and a reason string explaining why the creation is skipped.
+//
+// Scopes creation is skipped if:
+// - The flag for skipping scopes creation (`-skip-scopes-creation`) is provided.
+//
+// Returns:
+// - bool: True if the creation of scopes should be skipped, false otherwise.
+// - string: A reason string explaining why the creation is skipped, or an empty string if not skipped.
+func (f *initFlags) SkipScopesCreation() (bool, string) {
+	if f.flagSkipScopesCreation {
+		return true, reasonSkipScopesCreation
+	}
+	return false, ""
+}
+
+// SkipHostResourcesCreation checks if the creation of default host resources should be skipped.
+// It returns a boolean indicating whether to skip and a reason string explaining why the creation is skipped.
+//
+// Host resources creation is skipped if:
+// - The flag for skipping host resources creation (`-skip-host-resources-creation`) is provided.
+// - Any of the dependent resources were skipped.
+//
+// Dependent resources include:
+// - Scopes creation
+//
+// Returns:
+// - bool: True if the creation of host resources should be skipped, false otherwise.
+// - string: A reason string explaining why the creation is skipped, or an empty string if not skipped.
+func (f *initFlags) SkipHostResourcesCreation() (bool, string) {
+	if f.flagSkipHostResourcesCreation {
+		return true, reasonSkipHostResourcesCreation
+	}
+	if skip, reason := f.SkipScopesCreation(); skip {
+		return true, reason
+	}
+	return false, ""
```

Comment:
Nit, but you could simplify this to
```suggestion
	return f.SkipScopesCreation()
```